lib/_http_server.js

```diff
@@ -581,11 +581,7 @@ function socketOnTimeout() {
 
 function socketOnClose(socket, state) {
   debug('server socket close');
-  // Mark this parser as reusable
-  if (socket.parser) {
-    freeParser(socket.parser, null, socket);
```

Comment:
Just out of curiosity, in this handler, the socket will always have a parser because it must have been used, right?

Reply:
Not sure. But `freeParser` does all of these checks internally.

Reply:
Ok 😉